soc/nordic/common/uicr/gen_uicr.py

```diff
@@ -186,49 +178,27 @@ def main() -> None:
         uicr.VERSION.MAJOR = UICR_FORMAT_VERSION_MAJOR
         uicr.VERSION.MINOR = UICR_FORMAT_VERSION_MINOR
 
-        kconfig_str = args.in_config.read()
-        kconfig = parse_kconfig(kconfig_str)
-
-        edt = pickle.load(args.in_edt_pickle)
+        # Create a single hex object that will contain both UICR and periphconf data
+        merged_hex = IntelHex()
 
-        try:
-            periphconf_partition = edt.label2node[PERIPHCONF_NODELABEL]
-        except LookupError as e:
-            raise ScriptError(
-                "Failed to find a PERIPHCONF partition in the devicetree. "
-                f"Expected a DT node with label '{PERIPHCONF_NODELABEL}'."
-            ) from e
+        if args.in_periphconf_elfs:  # Check if periphconf data is provided
+            periphconf_combined = extract_and_combine_periphconfs(args.in_periphconf_elfs)
 
-        flash_base_address = periphconf_partition.flash_controller.regs[0].addr
-        periphconf_address = flash_base_address + periphconf_partition.regs[0].addr
-        periphconf_size = periphconf_partition.regs[0].size
+            padding_len = args.periphconf_size - len(periphconf_combined)
+            periphconf_final = periphconf_combined + bytes([0xFF for _ in range(padding_len)])
 
-        periphconf_combined = extract_and_combine_periphconfs(args.in_periphconf_elfs)
-        padding_len = periphconf_size - len(periphconf_combined)
-        periphconf_final = periphconf_combined + bytes([0xFF for _ in range(padding_len)])
+            # Add periphconf data to the merged hex
+            merged_hex.frombytes(periphconf_final, offset=args.periphconf_address)
 
-        if kconfig.get("CONFIG_NRF_HALTIUM_UICR_PERIPHCONF") == "y":
             uicr.PERIPHCONF.ENABLE = ENABLED_VALUE
-            uicr.PERIPHCONF.ADDRESS = periphconf_address
-            uicr.PERIPHCONF.MAXCOUNT = math.floor(periphconf_size / 8)
-
-        try:
-            uicr_node = edt.label2node[UICR_NODELABEL]
-        except LookupError as e:
-            raise ScriptError(
-                "Failed to find UICR node in the devicetree. "
-                f"Expected a DT node with label '{UICR_NODELABEL}'."
-            ) from e
+            uicr.PERIPHCONF.ADDRESS = args.periphconf_address
+            uicr.PERIPHCONF.MAXCOUNT = math.floor(args.periphconf_size / 8)
```

Comment:
nit: noticed this earlier, but this could just be `args.periphconf_size // 8`

Reply:
How about we error-out when it's not divisible by 8?